### ISSUE TITLE
bug: tx sweep failing on already-spent nullifier

### ISSUE THREAD

**Issue by aubrika** [penumbra-zone/penumbra#4574]
**Describe the bug**

It seems `tx sweep` is currently broken, as reported here in Discord:

https://discord.com/channels/824484045370818580/1172730682783563826/1248651765323137178

**Comment by zbuc** [comment]
I am able to `tx sweep` (mostly) successfully, however @conorsch and I noted that there seems to possibly be intermittent view server corruption leading to errors like this and SCT hash mismatches.

I say (mostly) successfully because I did get a different error...

```
# truncated: 24 successful sweeps

building sweep 24 of 25
building transaction [9 actions, 9 proofs]...
finished proving in 3.136 seconds [9 actions, 9 proofs, 4405 bytes]
broadcasting transaction and awaiting confirmation...
transaction broadcast successfully: 52cf05c1d14fa7791f44f6e24596376262a884e6718b97220c2dbc019bb4c326
transaction confirmed and detected: 52cf05c1d14fa7791f44f6e24596376262a884e6718b97220c2dbc019bb4c326
building sweep 0 of 3
building transaction [11 actions, 11 proofs]...
finished proving in 3.587 seconds [11 actions, 11 proofs, 5459 bytes]
broadcasting transaction and awaiting confirmation...
transaction broadcast successfully: bb88032a4eed65564973dd9aa7956113bf49f6469ca25d781534517f5bfecb76
transaction confirmed and detected: bb88032a4eed65564973dd9aa7956113bf49f6469ca25d781534517f5bfecb76 @ height 434808
building sweep 1 of 3
building transaction [9 actions, 9 proofs]...
Error: status: InvalidArgument, message: "Note commitment missing", details: [], metadata: MetadataMap { headers: {"content-type": "application/grpc"} }
```

**Comment by cratelyn** [comment]
x-ref #4601

**Comment by cratelyn** [comment]
#4605 introduced test coverage for this. #4633 disabled that test, due to the issues outlined in #4517. once we have support for an in-memory view service in tests, we can re-enable that.

**Comment by cratelyn** [comment]
for visibility, cross-posting this comment: https://github.com/penumbra-zone/penumbra/issues/4601#issuecomment-2186961948

**Comment by aubrika** [comment]
Closed as complete, should be fixed by #4578
